STM32Cube/Tasks/log.c

```diff
@@ -189,10 +193,45 @@ bool logDebug(const LogDataSource_t source, const char* msg, ...)
 #endif
 }
 
+static int computeFolderSize(const char *path) {
+	DIR dir;
+	FILINFO fno;
+	int nfile = 0;
+
+	if (f_opendir(&dir, path) == FR_OK) {
+		while (f_readdir(&dir, &fno) == FR_OK && fno.fname[0]) {
+			if (!(fno.fattrib & AM_DIR))
+				++nfile;
+		}
+		f_closedir(&dir);
+	}
+
+	return nfile;
+}
+
+static void initUniqueLogFileName(const char *logsPath) {
+	int nextValidFileNumber = computeFolderSize(logsPath) - 1;
+	while (snprintf(logFileName, sizeof(logFileName), "%s/%d.txt", logsPath,
+				 ++nextValidFileNumber) > 0 &&
+		f_stat(logFileName, NULL) == FR_OK);
+}
+
 // ----------------------------------------------------------------------------
 
 void logTask(void *argument)
 {
+	// initalize log file stuff
+	FATFS fs;
+	(void)f_mount(&fs, "", 0);
```

Comment:
what does `(void)` do?

Reply:
`void` just meant that I was purposely avoiding checking the return value because if it's failing it's going to fail and I don't know what we're supposed to do if it fails